File: trialcode/loop.js
timer=timer=require("ringo/utils/debug").timer;
radians = java.lang.Math.toRadians;

var cc90 = radians(90);
var cw90 = radians(-90);

function draw(g2d){
  var n = 10000, len = 3;
  print("n: " + n + " len: " + len);
  for(var i=0;i<n;i++){
    g2d.drawLine(0, 0, len, 0);
    g2d.translate(len,0);
    g2d.rotate(cc90);
    g2d.drawLine(0, 0, len, 0);
    g2d.translate(len,0);
    g2d.rotate(cw90);
  }
}


width=800;
height=800;

frame = new javax.swing.JFrame();
pan = new JavaAdapter(javax.swing.JPanel,
  {
    paint:function(g2d){
      timer(function(){draw(g2d)});
  }
})

frame.title="Ringo";
frame.setContentPane(pan);
frame.pack();
frame.setSize(width,height);
frame.show();

File: trialcode/loop.clj
(defn radians [degrees] (. java.lang.Math toRadians degrees))

(def cc90 (radians 90))
(def cw90 (radians -90))

(defn draw [^java.awt.Graphics2D g2d]
  (let [n 10000
	len 3]
    (dotimes [i n]
      (doto g2d
	(.drawLine 0 0 len 0)
	(.translate len 0)
	(.rotate cc90)
	(.drawLine 0 0 len 0)
	(.translate len 0)
	(.rotate cw90)))))

(def width 800)
(def height 800)

(def frame (new javax.swing.JFrame))
(def pan (proxy [javax.swing.JPanel] []
  (paint [g2d]
    (time (draw g2d))
  )
))

(doto frame
  (.setTitle "Clojure")
  (.setContentPane pan)
  (.pack)
  (.setSize width height)
  (.show))

